lms/djangoapps/course_api/api.py

```diff
@@ -100,13 +100,10 @@ def _filter_by_search(course_queryset, search_term):
     )
 
     search_courses_ids = {course['data']['id'] for course in search_courses['results']}
-
+    courses = [course for course in course_queryset if str(course.id) in search_courses_ids]
```

Comment:
@ormsbee is there gonna be a performance impact for changing the lazy sequence this way?  I feel like getting an inaccurate count here is expected

https://github.com/openedx/edx-platform/blob/2033dcf6ace133719aaeb72dc5dd6ee521a7ac42/openedx/core/lib/api/view_utils.py#L246C1-L248C47

> It is immutable, and accepts an estimated length in order to support `__len__` without exhausting the underlying sequence.

Reply:
@feanil: Almost certainly, yes. The whole point of `LazySequence` is to not have to iterate the entire queryset, which could potentially be 10K courses.

If this is related to the mobile effort, would it be possible to cap the number of results (e.g. pass in a parameter that will tell it to return no more than X results)? Doing something like the above with a cap of 100 courses or something should be fine.

Reply:
@oksana-slu let me know your thoughts but given the potential performance problem, I'm reluctant to land the change as is,.

Reply:
Wow, that's definitely a miss.
We discussed the possible solution today with @volodymyr-chekyrta, and it's a fair tradeoff to limit the number of returned results for mobile to 100 or even to 50 for now.